src/snap/metric.cc

```diff
@@ -263,3 +276,96 @@ void Metric::Metric::set_last_advert_tm(system_clock::time_point tp) {
     tm->set_sec(nanos / std::nano::den);
     tm->set_nsec(nanos % std::nano::den);
 }
+
+
+
+Namespace::Namespace(std::vector<std::string> ns){
+    
+    for (auto &string_iterator : ns){
+        this->namespace_elements.push_back(Namespace_Element(string_iterator));
+    }
+}
+Namespace::Namespace(){}
+
+Namespace::~Namespace(){
+
+}
+const Namespace_Element Namespace::operator[] (int index) const{
```

Comment:
Please do not use underscore while using camelcase notation, consider calling it just `NamespaceElement`.